### PR TITLE
Add country flags to language select

### PR DESCRIPTION
**Pull request by jguthrie100**
I thought it might be nice to include country flags within the language select dropdown..

image

### CODE REVIEW

**Comment by tananaev** [comment]
1. How do we deal with languages that are used in multiple countries?
2. Can we use Unicode symbols instead of including another library?

**Comment by jguthrie100** [comment]
Sure, I can change it to use emojis/unicode.

For multiple countries, I just used the main country (apart from English, where I used the UK, but could make it USA)

I was thinking of even listing multiple flags for certain languages, but it might look a bit cluttered!

**Comment by tananaev** [comment]
Yeah, let's do emojis and set the English to the USA flag as the more common.

**Comment by jguthrie100** [comment]
Apparently Windows doesn't have support for flag emojis? I don't have windows so I can't check..

https://answers.microsoft.com/en-us/windows/forum/all/flag-emoji/85b163bc-786a-4918-9042-763ccf4b6c05

**Comment by tananaev** [comment]
It's not working on windows even with a custom font, like roboto?

**Comment by jguthrie100** [comment]
My understanding that Microsoft have left all unicode flags out of Windows in order to avoid issues around supporting various countries (i.e. whether to include Palestine/Israel flags etc)

**Comment by tananaev** [comment]
Can you update the pr and I'll try.

File: modern/src/common/components/LocalizationProvider.js [hunk 1]
```diff
@@ -64,62 +64,62 @@ import zhTW from '../../resources/l10n/zh_TW.json';
...
-  sv: { data: sv, name: 'Svenska' },
-  ta: { data: ta, name: 'தமிழ்' },
-  th: { data: th, name: 'ไทย' },
-  tr: { data: tr, name: 'Türkçe' },
-  uk: { data: uk, name: 'Українська' },
-  uz: { data: uz, name: 'Oʻzbekcha' },
-  vi: { data: vi, name: 'Tiếng Việt' },
-  zh: { data: zh, name: '中文' },
-  zhTW: { data: zhTW, name: '中文 (Taiwan)' },
+  af: { data: af, countryCode: 'ZA', name: 'Afrikaans' },
```
Comment (tananaev) [inline modern/src/common/components/LocalizationProvider.js]: Let's rename `countryCode` to just `country`.

File: modern/src/common/components/LocalizationProvider.jsx [hunk 2]
```diff
@@ -64,62 +64,62 @@ import zhTW from '../../resources/l10n/zh_TW.json';
...
+  km: { data: km, countryCode: 'KH', name: 'ភាសាខ្មែរ' },
+  lo: { data: lo, countryCode: 'LA', name: 'ລາວ' },
+  lt: { data: lt, countryCode: 'LT', name: 'Lietuvių' },
+  lv: { data: lv, countryCode: 'LV', name: 'Latviešu' },
+  ml: { data: ml, countryCode: 'IN', name: 'മലയാളം' },
+  mn: { data: mn, countryCode: 'MN', name: 'Монгол хэл' },
+  ms: { data: ms, countryCode: 'MY', name: 'بهاس ملايو' },
+  ne: { data: ne, countryCode: 'NP', name: 'नेपाली' },
+  nl: { data: nl, countryCode: 'NL', name: 'Nederlands' },
+  nb: { data: nb, countryCode: 'NO', name: 'Norsk bokmål' },
```
Comment (tananaev) [inline modern/src/common/components/LocalizationProvider.jsx:102]: Please keep the order as before.

File: modern/src/common/components/LocalizationProvider.jsx [hunk 3]
```diff
@@ -186,3 +186,11 @@ export const useTranslationKeys = (predicate) => {
   const { data } = context.languages[context.language];
   return Object.keys(data).filter(predicate);
 };
+
+export const getFlagEmoji = (countryCode) => {
+  const codePoints = countryCode
+    .toUpperCase()
```
Comment (tananaev) [inline modern/src/common/components/LocalizationProvider.jsx:192]: I think this is not needed.

**Comment by jguthrie100** [comment]
Have you tested it on Windows? I just heard back from someone that it only shows the unicode letters..

**Comment by tananaev** [comment]
Doesn't work on Windows, but I've added this check: https://github.com/traccar/traccar-web/commit/c36d1bed4a8781b2307fe2bfcdd1c6d4ddd00d64.

**Comment by jguthrie100** [comment]
Is it not worth just including the tiny library so the majority Windows users can see flags too?

**Comment by tananaev** [comment]
OK, let's do it.

**Pull request merged**